src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -664,5 +712,7 @@ protected override XmlObjectSerialization EnsureXmlSerialization()
 
             return _context.BaseLibrary.FindTypeProviderForSchema(defaultDerivedSchema) as ObjectType;
         }
+
+        private record SchemaObjectTypeFields(IReadOnlyList<Parameter> PublicConstructorParameters, IReadOnlyList<Parameter> SerializationParameters);
```

Comment:
We are taking steps to migrate this therefore I just mimick the interfaces like IObjectTypeFields but do not really implement it because to implement this interface more changes are required.